### PR TITLE
Feat: Project managed settings files

### PR DESCRIPTION
**Pull request by jwagantall**
Add the ability for projects to manage
Jenkins configuration files with JCasC.

Signed-off-by: Jessica Wagantall <jwagantall@linuxfoundation.org>

### CODE REVIEW

File: jenkins-config/managed-config-files/globalMavenSettings/global-settings/content [hunk 1]
```diff
@@ -0,0 +1,257 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (tykeal) [inline jenkins-config/managed-config-files/globalMavenSettings/global-settings/content:2]: You can rip out all of the boiler plate comments. They aren't needed

File: jenkins-config/managed-config-files/mavenSettings/dentos-settings/content [hunk 2]
```diff
@@ -0,0 +1,257 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (tykeal) [inline jenkins-config/managed-config-files/mavenSettings/dentos-settings/content:2]: You should rip out all the boiler plate comments as well as all of the sections for this file. It should end up being the <?xml header line and <settings open and </settings> close

Also, move one out of the managed-config-files directory to a directory called managed-config-templates and name it mavenSettings-content and then symlink to it for all of the other mavenSettings content files.

File: jenkins-config/managed-config-files/mavenSettings/dentos-settings/serverCredentialMappings.sandbox.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (tykeal) [inline jenkins-config/managed-config-files/mavenSettings/dentos-settings/serverCredentialMappings.sandbox.yaml:1]: Move this to the template directory I specified above and symlink to that new file

File: jenkins-config/managed-config-files/mavenSettings/jenkins-log-archives-settings/content [hunk 4]
```diff
@@ -0,0 +1,257 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tykeal) [inline jenkins-config/managed-config-files/mavenSettings/jenkins-log-archives-settings/content:1]: See above comment

**Review by tykeal — CHANGES_REQUESTED** [review]

File: jenkins-config/managed-config-files/globalMavenSettings/global-settings/content (hunk 1 above)
Comment (jwagantall) [reply jenkins-config/managed-config-files/globalMavenSettings/global-settings/content:2]: Thank you .. done..

File: jenkins-config/managed-config-files/mavenSettings/dentos-settings/content (hunk 2 above)
Comment (jwagantall) [reply jenkins-config/managed-config-files/mavenSettings/dentos-settings/content:2]: got it! thanks!

File: jenkins-config/managed-config-files/mavenSettings/jenkins-log-archives-settings/content (hunk 4 above)
Comment (jwagantall) [reply jenkins-config/managed-config-files/mavenSettings/jenkins-log-archives-settings/content:1]: done

File: jenkins-config/managed-config-files/mavenSettings/dentos-settings/serverCredentialMappings.sandbox.yaml (hunk 3 above)
Comment (jwagantall) [reply jenkins-config/managed-config-files/mavenSettings/dentos-settings/serverCredentialMappings.sandbox.yaml:1]: Done.. thank you

File: jenkins-config/managed-config-files/globalMavenSettings/global-settings/content [hunk 5]
```diff
@@ -0,0 +1 @@
+../../../managed-config-templates/mavenSettings-content
```
Comment (tykeal) [inline jenkins-config/managed-config-files/globalMavenSettings/global-settings/content:1]: globalMavenSettings content is not the same as the standard mavenSettings content. You should take the regular boiler plate globalMavenSettings and remove all comments, but all of the sections that it had should remain (we've never tested to see what happens if they're removed!)

**Review by tykeal — CHANGES_REQUESTED** [review]

File: jenkins-config/managed-config-files/globalMavenSettings/global-settings/content (hunk 5 above)
Comment (jwagantall) [reply jenkins-config/managed-config-files/globalMavenSettings/global-settings/content:1]: done

**Review by tykeal — APPROVED** [review]

**Review by MightyNerdEric — APPROVED** [review]

**Pull request merged**